{
    "issue": {
        "title": "[Python] cannot import pip installed pyarrow on OS X (10.9)",
        "body": "***Note**: This issue was originally created as [ARROW-2326](https://issues.apache.org/jira/browse/ARROW-2326). Please see the [migration documentation](https://gist.github.com/toddfarmer/12aa88361532d21902818a6044fda4c3) for further details.*\n\n### Original Issue Description:\n```java\n\r\n$ pip3 install pyarrow --user\r\nCollecting pyarrow\r\nUsing cached pyarrow-0.8.0-cp36-cp36m-macosx_10_6_intel.whl\r\nRequirement already satisfied: six>=1.0.0 in ./Library/Python/3.6/lib/python/site-packages (from pyarrow)\r\nCollecting numpy>=1.10 (from pyarrow)\r\nUsing cached numpy-1.14.2-cp36-cp36m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl\r\nInstalling collected packages: numpy, pyarrow\r\nSuccessfully installed numpy-1.14.2 pyarrow-0.8.0\r\n\r\n$ python3\r\nPython 3.6.1 (v3.6.1:69c0db5050, Mar 21 2017, 01:21:04) \r\n[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin\r\nType \"help\", \"copyright\", \"credits\" or \"license\" for more information.\r\n>>> import pyarrow\r\nTraceback (most recent call last):\r\nFile \"<stdin>\", line 1, in <module>\r\nFile \"/Users/pi/Library/Python/3.6/lib/python/site-packages/pyarrow/__init__.py\", line 32, in <module>\r\nfrom pyarrow.lib import cpu_count, set_cpu_count\r\nImportError: dlopen(/Users/pi/Library/Python/3.6/lib/python/site-packages/pyarrow/lib.cpython-36m-darwin.so, 2): Library not loaded: @rpath/libarrow.0.dylib\r\nReferenced from: /Users/pi/Library/Python/3.6/lib/python/site-packages/pyarrow/lib.cpython-36m-darwin.so\r\nReason: image not found\r\n```\r\nI dug into it a bit and found that in older versions of install.rst, Wes mentioned that XCode 6 had trouble with rpath, so not sure if that's what's going on here for me. I'm on 10.9, I know it's really old, so if these wheels can't be made to run on my ancient OS, I just wanted to report this so the wheels uploaded to PyPI can reflect this incompatibility, if that is indeed the case. I might also try some otool / install_name_tool tomfoolery to see if I can get a workaround for myself.\r\n\r\nThank you!",
        "created_at": "2018-03-17T06:16:21.000Z",
        "updated_at": "2018-07-19T12:47:09.000Z",
        "labels": [
            "Migrated from Jira",
            "Component: Python",
            "Type: bug"
        ],
        "closed": true,
        "closed_at": "2018-07-19T07:49:30.000Z"
    },
    "comments": [
        {
            "created_at": "2018-03-19T22:12:09.594Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16405536) by Wes McKinney (wesm):*\nhi `[~pi]` \u2013 these wheels should definitely work on OS 10.9. Do any of the older wheels (0.5 through 0.7.1) work for you, or have these always been broken? Marked for 0.10.0 (though if we have a 0.9.1 bugfix release then we can get this fixed sooner)"
        },
        {
            "created_at": "2018-03-23T01:08:28.549Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16410654) by Paul Ivanov (pi):*\nYeah - none of those work here, neither in python 2.7 nor in 3.6, so I guess they've always been broken.\r\n\r\n\u00a0\r\n```java\n\r\n$ for x in 0.5.0 0.6.0 0.7.0 0.7.1 0.8.0; do mkvirtualenv --python=python3 arrow$x; pip install pyarrow==$x; python -c \"import pyarrow as pa\"; deactivate; echo \"Done with $x\"\r\n; done\r\nRunning virtualenv with interpreter /Library/Frameworks/Python.framework/Versions/3.6/bin/python3\r\nUsing base prefix '/Library/Frameworks/Python.framework/Versions/3.6'\r\nNew python executable in /Users/pi/.virtualenvs/arrow0.5.0/bin/python3\r\nAlso creating executable in /Users/pi/.virtualenvs/arrow0.5.0/bin/python\r\nInstalling setuptools, pip, wheel...done.\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.5.0/bin/predeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.5.0/bin/postdeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.5.0/bin/preactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.5.0/bin/postactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.5.0/bin/get_env_details\r\nCollecting pyarrow==0.5.0\r\nDownloading pyarrow-0.5.0-cp36-cp36m-macosx_10_6_intel.whl (6.1MB)\r\n100% |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 6.1MB 180kB/s\r\nCollecting six>=1.0.0 (from pyarrow==0.5.0)\r\nUsing cached six-1.11.0-py2.py3-none-any.whl\r\nCollecting numpy>=1.10 (from pyarrow==0.5.0)\r\nUsing cached numpy-1.14.2-cp36-cp36m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl\r\nInstalling collected packages: six, numpy, pyarrow\r\nSuccessfully installed numpy-1.14.2 pyarrow-0.5.0 six-1.11.0\r\nTraceback (most recent call last):\r\nFile \"<string>\", line 1, in <module>\r\nFile \"/Users/pi/.virtualenvs/arrow0.5.0/lib/python3.6/site-packages/pyarrow/__init__.py\", line 32, in <module>\r\nfrom pyarrow.lib import cpu_count, set_cpu_count\r\nImportError: dlopen(/Users/pi/.virtualenvs/arrow0.5.0/lib/python3.6/site-packages/pyarrow/lib.cpython-36m-darwin.so, 2): Library not loaded: @rpath/libarrow.0.dylib\r\nReferenced from: /Users/pi/.virtualenvs/arrow0.5.0/lib/python3.6/site-packages/pyarrow/lib.cpython-36m-darwin.so\r\nReason: image not found\r\nDone with 0.5.0\r\nRunning virtualenv with interpreter /Library/Frameworks/Python.framework/Versions/3.6/bin/python3\r\nUsing base prefix '/Library/Frameworks/Python.framework/Versions/3.6'\r\nNew python executable in /Users/pi/.virtualenvs/arrow0.6.0/bin/python3\r\nAlso creating executable in /Users/pi/.virtualenvs/arrow0.6.0/bin/python\r\nInstalling setuptools, pip, wheel...done.\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.6.0/bin/predeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.6.0/bin/postdeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.6.0/bin/preactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.6.0/bin/postactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.6.0/bin/get_env_details\r\nCollecting pyarrow==0.6.0\r\nDownloading pyarrow-0.6.0-cp36-cp36m-macosx_10_6_intel.whl (5.9MB)\r\n100% |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 5.9MB 188kB/s\r\nCollecting numpy>=1.10 (from pyarrow==0.6.0)\r\nUsing cached numpy-1.14.2-cp36-cp36m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl\r\nCollecting six>=1.0.0 (from pyarrow==0.6.0)\r\nUsing cached six-1.11.0-py2.py3-none-any.whl\r\nInstalling collected packages: numpy, six, pyarrow\r\nSuccessfully installed numpy-1.14.2 pyarrow-0.6.0 six-1.11.0\r\nTraceback (most recent call last):\r\nFile \"<string>\", line 1, in <module>\r\nFile \"/Users/pi/.virtualenvs/arrow0.6.0/lib/python3.6/site-packages/pyarrow/__init__.py\", line 32, in <module>\r\nfrom pyarrow.lib import cpu_count, set_cpu_count\r\nImportError: dlopen(/Users/pi/.virtualenvs/arrow0.6.0/lib/python3.6/site-packages/pyarrow/lib.cpython-36m-darwin.so, 2): Library not loaded: @rpath/libarrow.0.dylib\r\nReferenced from: /Users/pi/.virtualenvs/arrow0.6.0/lib/python3.6/site-packages/pyarrow/lib.cpython-36m-darwin.so\r\nReason: image not found\r\nDone with 0.6.0\r\nRunning virtualenv with interpreter /Library/Frameworks/Python.framework/Versions/3.6/bin/python3\r\nUsing base prefix '/Library/Frameworks/Python.framework/Versions/3.6'\r\nNew python executable in /Users/pi/.virtualenvs/arrow0.7.0/bin/python3\r\nAlso creating executable in /Users/pi/.virtualenvs/arrow0.7.0/bin/python\r\nInstalling setuptools, pip, wheel...done.\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.0/bin/predeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.0/bin/postdeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.0/bin/preactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.0/bin/postactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.0/bin/get_env_details\r\nCollecting pyarrow==0.7.0\r\nDownloading pyarrow-0.7.0-cp36-cp36m-macosx_10_6_intel.whl (6.3MB)\r\n100% |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 6.3MB 171kB/s\r\nCollecting six>=1.0.0 (from pyarrow==0.7.0)\r\nUsing cached six-1.11.0-py2.py3-none-any.whl\r\nCollecting numpy>=1.10 (from pyarrow==0.7.0)\r\nUsing cached numpy-1.14.2-cp36-cp36m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl\r\nInstalling collected packages: six, numpy, pyarrow\r\nSuccessfully installed numpy-1.14.2 pyarrow-0.7.0 six-1.11.0\r\nTraceback (most recent call last):\r\nFile \"<string>\", line 1, in <module>\r\nFile \"/Users/pi/.virtualenvs/arrow0.7.0/lib/python3.6/site-packages/pyarrow/__init__.py\", line 32, in <module>\r\nfrom pyarrow.lib import cpu_count, set_cpu_count\r\nImportError: dlopen(/Users/pi/.virtualenvs/arrow0.7.0/lib/python3.6/site-packages/pyarrow/lib.cpython-36m-darwin.so, 2): Library not loaded: @rpath/libarrow.0.dylib\r\nReferenced from: /Users/pi/.virtualenvs/arrow0.7.0/lib/python3.6/site-packages/pyarrow/lib.cpython-36m-darwin.so\r\nReason: image not found\r\nDone with 0.7.0\r\nRunning virtualenv with interpreter /Library/Frameworks/Python.framework/Versions/3.6/bin/python3\r\nUsing base prefix '/Library/Frameworks/Python.framework/Versions/3.6'\r\nNew python executable in /Users/pi/.virtualenvs/arrow0.7.1/bin/python3\r\nAlso creating executable in /Users/pi/.virtualenvs/arrow0.7.1/bin/python\r\nInstalling setuptools, pip, wheel...done.\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.1/bin/predeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.1/bin/postdeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.1/bin/preactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.1/bin/postactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.1/bin/get_env_details\r\nCollecting pyarrow==0.7.1\r\nDownloading pyarrow-0.7.1-cp36-cp36m-macosx_10_6_intel.whl (6.4MB)\r\n100% |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 6.4MB 162kB/s\r\nCollecting six>=1.0.0 (from pyarrow==0.7.1)\r\nUsing cached six-1.11.0-py2.py3-none-any.whl\r\nCollecting numpy>=1.10 (from pyarrow==0.7.1)\r\nUsing cached numpy-1.14.2-cp36-cp36m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl\r\nInstalling collected packages: six, numpy, pyarrow\r\nSuccessfully installed numpy-1.14.2 pyarrow-0.7.1 six-1.11.0\r\nTraceback (most recent call last):\r\nFile \"<string>\", line 1, in <module>\r\nFile \"/Users/pi/.virtualenvs/arrow0.7.1/lib/python3.6/site-packages/pyarrow/__init__.py\", line 32, in <module>\r\nfrom pyarrow.lib import cpu_count, set_cpu_count\r\nImportError: dlopen(/Users/pi/.virtualenvs/arrow0.7.1/lib/python3.6/site-packages/pyarrow/lib.cpython-36m-darwin.so, 2): Library not loaded: @rpath/libarrow.0.dylib\r\nReferenced from: /Users/pi/.virtualenvs/arrow0.7.1/lib/python3.6/site-packages/pyarrow/lib.cpython-36m-darwin.so\r\nReason: image not found\r\nDone with 0.7.1\r\nRunning virtualenv with interpreter /Library/Frameworks/Python.framework/Versions/3.6/bin/python3\r\nUsing base prefix '/Library/Frameworks/Python.framework/Versions/3.6'\r\nNew python executable in /Users/pi/.virtualenvs/arrow0.8.0/bin/python3\r\nAlso creating executable in /Users/pi/.virtualenvs/arrow0.8.0/bin/python\r\nInstalling setuptools, pip, wheel...done.\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.8.0/bin/predeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.8.0/bin/postdeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.8.0/bin/preactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.8.0/bin/postactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.8.0/bin/get_env_details\r\nCollecting pyarrow==0.8.0\r\nUsing cached pyarrow-0.8.0-cp36-cp36m-macosx_10_6_intel.whl\r\nCollecting six>=1.0.0 (from pyarrow==0.8.0)\r\nUsing cached six-1.11.0-py2.py3-none-any.whl\r\nCollecting numpy>=1.10 (from pyarrow==0.8.0)\r\nUsing cached numpy-1.14.2-cp36-cp36m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl\r\nInstalling collected packages: six, numpy, pyarrow\r\nSuccessfully installed numpy-1.14.2 pyarrow-0.8.0 six-1.11.0\r\nTraceback (most recent call last):\r\nFile \"<string>\", line 1, in <module>\r\nFile \"/Users/pi/.virtualenvs/arrow0.8.0/lib/python3.6/site-packages/pyarrow/__init__.py\", line 32, in <module>\r\nfrom pyarrow.lib import cpu_count, set_cpu_count\r\nImportError: dlopen(/Users/pi/.virtualenvs/arrow0.8.0/lib/python3.6/site-packages/pyarrow/lib.cpython-36m-darwin.so, 2): Library not loaded: @rpath/libarrow.0.dylib\r\nReferenced from: /Users/pi/.virtualenvs/arrow0.8.0/lib/python3.6/site-packages/pyarrow/lib.cpython-36m-darwin.so\r\nReason: image not found\r\nDone with 0.8.0\r\n\r\n\u00a0\r\n\r\n$ for x in 0.5.0 0.6.0 0.7.0 0.7.1 0.8.0; do mkvirtualenv --python=python2 arrow$x-py2; pip install pyarrow==$x; python -c \"import pyarrow as pa\"; deactivate; echo \"Done with\r\n$x\" ; done\r\nRunning virtualenv with interpreter /usr/local/bin/python2\r\nNew python executable in /Users/pi/.virtualenvs/arrow0.5.0-py2/bin/python2.7\r\nAlso creating executable in /Users/pi/.virtualenvs/arrow0.5.0-py2/bin/python\r\nInstalling setuptools, pip, wheel...done.\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.5.0-py2/bin/predeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.5.0-py2/bin/postdeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.5.0-py2/bin/preactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.5.0-py2/bin/postactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.5.0-py2/bin/get_env_details\r\nCollecting pyarrow==0.5.0\r\nDownloading pyarrow-0.5.0-cp27-cp27m-macosx_10_6_intel.whl (6.1MB)\r\n100% |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 6.1MB 149kB/s\r\nCollecting numpy>=1.10 (from pyarrow==0.5.0)\r\nDownloading numpy-1.14.2-cp27-cp27m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl (4.7MB)\r\n100% |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 4.7MB 209kB/s\r\nCollecting six>=1.0.0 (from pyarrow==0.5.0)\r\nUsing cached six-1.11.0-py2.py3-none-any.whl\r\nInstalling collected packages: numpy, six, pyarrow\r\nSuccessfully installed numpy-1.14.2 pyarrow-0.5.0 six-1.11.0\r\nTraceback (most recent call last):\r\nFile \"<string>\", line 1, in <module>\r\nFile \"/Users/pi/.virtualenvs/arrow0.5.0-py2/lib/python2.7/site-packages/pyarrow/__init__.py\", line 32, in <module>\r\nfrom pyarrow.lib import cpu_count, set_cpu_count\r\nImportError: dlopen(/Users/pi/.virtualenvs/arrow0.5.0-py2/lib/python2.7/site-packages/pyarrow/lib.so, 2): Library not loaded: @rpath/libarrow.0.dylib\r\nReferenced from: /Users/pi/.virtualenvs/arrow0.5.0-py2/lib/python2.7/site-packages/pyarrow/lib.so\r\nReason: image not found\r\nDone with 0.5.0\r\nRunning virtualenv with interpreter /usr/local/bin/python2\r\nNew python executable in /Users/pi/.virtualenvs/arrow0.6.0-py2/bin/python2.7\r\nAlso creating executable in /Users/pi/.virtualenvs/arrow0.6.0-py2/bin/python\r\nInstalling setuptools, pip, wheel...done.\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.6.0-py2/bin/predeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.6.0-py2/bin/postdeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.6.0-py2/bin/preactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.6.0-py2/bin/postactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.6.0-py2/bin/get_env_details\r\nCollecting pyarrow==0.6.0\r\nDownloading pyarrow-0.6.0-cp27-cp27m-macosx_10_6_intel.whl (5.9MB)\r\n100% |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 5.9MB 162kB/s\r\nCollecting numpy>=1.10 (from pyarrow==0.6.0)\r\nUsing cached numpy-1.14.2-cp27-cp27m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl\r\nCollecting six>=1.0.0 (from pyarrow==0.6.0)\r\nUsing cached six-1.11.0-py2.py3-none-any.whl\r\nInstalling collected packages: numpy, six, pyarrow\r\nSuccessfully installed numpy-1.14.2 pyarrow-0.6.0 six-1.11.0\r\nTraceback (most recent call last):\r\nFile \"<string>\", line 1, in <module>\r\nFile \"/Users/pi/.virtualenvs/arrow0.6.0-py2/lib/python2.7/site-packages/pyarrow/__init__.py\", line 32, in <module>\r\nfrom pyarrow.lib import cpu_count, set_cpu_count\r\nImportError: dlopen(/Users/pi/.virtualenvs/arrow0.6.0-py2/lib/python2.7/site-packages/pyarrow/lib.so, 2): Library not loaded: @rpath/libarrow.0.dylib\r\nReferenced from: /Users/pi/.virtualenvs/arrow0.6.0-py2/lib/python2.7/site-packages/pyarrow/lib.so\r\nReason: image not found\r\nDone with 0.6.0\r\nRunning virtualenv with interpreter /usr/local/bin/python2\r\nNew python executable in /Users/pi/.virtualenvs/arrow0.7.0-py2/bin/python2.7\r\nAlso creating executable in /Users/pi/.virtualenvs/arrow0.7.0-py2/bin/python\r\nInstalling setuptools, pip, wheel...done.\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.0-py2/bin/predeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.0-py2/bin/postdeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.0-py2/bin/preactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.0-py2/bin/postactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.0-py2/bin/get_env_details\r\nCollecting pyarrow==0.7.0\r\nDownloading pyarrow-0.7.0-cp27-cp27m-macosx_10_6_intel.whl (6.3MB)\r\n100% |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 6.3MB 145kB/s\r\nCollecting numpy>=1.10 (from pyarrow==0.7.0)\r\nUsing cached numpy-1.14.2-cp27-cp27m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl\r\nCollecting six>=1.0.0 (from pyarrow==0.7.0)\r\nUsing cached six-1.11.0-py2.py3-none-any.whl\r\nInstalling collected packages: numpy, six, pyarrow\r\nSuccessfully installed numpy-1.14.2 pyarrow-0.7.0 six-1.11.0\r\nTraceback (most recent call last):\r\nFile \"<string>\", line 1, in <module>\r\nFile \"/Users/pi/.virtualenvs/arrow0.7.0-py2/lib/python2.7/site-packages/pyarrow/__init__.py\", line 32, in <module>\r\nfrom pyarrow.lib import cpu_count, set_cpu_count\r\nImportError: dlopen(/Users/pi/.virtualenvs/arrow0.7.0-py2/lib/python2.7/site-packages/pyarrow/lib.so, 2): Library not loaded: @rpath/libarrow.0.dylib\r\nReferenced from: /Users/pi/.virtualenvs/arrow0.7.0-py2/lib/python2.7/site-packages/pyarrow/lib.so\r\nReason: image not found\r\nDone with 0.7.0\r\nRunning virtualenv with interpreter /usr/local/bin/python2\r\nNew python executable in /Users/pi/.virtualenvs/arrow0.7.1-py2/bin/python2.7\r\nAlso creating executable in /Users/pi/.virtualenvs/arrow0.7.1-py2/bin/python\r\nInstalling setuptools, pip, wheel...done.\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.1-py2/bin/predeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.1-py2/bin/postdeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.1-py2/bin/preactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.1-py2/bin/postactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.7.1-py2/bin/get_env_details\r\nCollecting pyarrow==0.7.1\r\nDownloading pyarrow-0.7.1-cp27-cp27m-macosx_10_6_intel.whl (6.4MB)\r\n100% |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 6.4MB 146kB/s\r\nCollecting numpy>=1.10 (from pyarrow==0.7.1)\r\nUsing cached numpy-1.14.2-cp27-cp27m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl\r\nCollecting six>=1.0.0 (from pyarrow==0.7.1)\r\nUsing cached six-1.11.0-py2.py3-none-any.whl\r\nInstalling collected packages: numpy, six, pyarrow\r\nSuccessfully installed numpy-1.14.2 pyarrow-0.7.1 six-1.11.0\r\nTraceback (most recent call last):\r\nFile \"<string>\", line 1, in <module>\r\nFile \"/Users/pi/.virtualenvs/arrow0.7.1-py2/lib/python2.7/site-packages/pyarrow/__init__.py\", line 32, in <module>\r\nfrom pyarrow.lib import cpu_count, set_cpu_count\r\nImportError: dlopen(/Users/pi/.virtualenvs/arrow0.7.1-py2/lib/python2.7/site-packages/pyarrow/lib.so, 2): Library not loaded: @rpath/libarrow.0.dylib\r\nReferenced from: /Users/pi/.virtualenvs/arrow0.7.1-py2/lib/python2.7/site-packages/pyarrow/lib.so\r\nReason: image not found\r\nDone with 0.7.1\r\nRunning virtualenv with interpreter /usr/local/bin/python2\r\nNew python executable in /Users/pi/.virtualenvs/arrow0.8.0-py2/bin/python2.7\r\nAlso creating executable in /Users/pi/.virtualenvs/arrow0.8.0-py2/bin/python\r\nInstalling setuptools, pip, wheel...done.\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.8.0-py2/bin/predeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.8.0-py2/bin/postdeactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.8.0-py2/bin/preactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.8.0-py2/bin/postactivate\r\nvirtualenvwrapper.user_scripts creating /Users/pi/.virtualenvs/arrow0.8.0-py2/bin/get_env_details\r\nCollecting pyarrow==0.8.0\r\nDownloading pyarrow-0.8.0-cp27-cp27m-macosx_10_6_intel.whl (6.7MB)\r\n100% |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 6.7MB 145kB/s\r\nCollecting futures (from pyarrow==0.8.0)\r\nDownloading futures-3.2.0-py2-none-any.whl\r\nCollecting numpy>=1.10 (from pyarrow==0.8.0)\r\nUsing cached numpy-1.14.2-cp27-cp27m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl\r\nCollecting six>=1.0.0 (from pyarrow==0.8.0)\r\nUsing cached six-1.11.0-py2.py3-none-any.whl\r\nInstalling collected packages: futures, numpy, six, pyarrow\r\nSuccessfully installed futures-3.2.0 numpy-1.14.2 pyarrow-0.8.0 six-1.11.0\r\nTraceback (most recent call last):\r\nFile \"<string>\", line 1, in <module>\r\nFile \"/Users/pi/.virtualenvs/arrow0.8.0-py2/lib/python2.7/site-packages/pyarrow/__init__.py\", line 32, in <module>\r\nfrom pyarrow.lib import cpu_count, set_cpu_count\r\nImportError: dlopen(/Users/pi/.virtualenvs/arrow0.8.0-py2/lib/python2.7/site-packages/pyarrow/lib.so, 2): Library not loaded: @rpath/libarrow.0.dylib\r\nReferenced from: /Users/pi/.virtualenvs/arrow0.8.0-py2/lib/python2.7/site-packages/pyarrow/lib.so\r\nReason: image not found\r\nDone with 0.8.0\r\n```\r\n\u00a0"
        },
        {
            "created_at": "2018-03-23T01:13:48.826Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16410658) by Wes McKinney (wesm):*\nI just wrote an e-mail to Matthew Brett and Nathaniel Smith who know a lot more about this than I do. Hopefully they will chime in here, or I can relay their advice about how to diagnose/fix the issue back here"
        },
        {
            "created_at": "2018-03-23T07:40:26.054Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16410924) by Matthew Brett (matthewbrett):*\nMy guess is that somewhere in the build process pyarrow is using rpath to point to the library it needs, and older delocate did not look at the rpath, so does not find and vendor the library.\u00a0 My guess is that current delocate will generate a correct wheel - is that true?\u00a0 Otherwise, surgery could help - unpack the wheel before running delocate, use otool to point the relevant file to the libarrow library full path (rather than using rpath), and pack up again, before running delocate."
        },
        {
            "created_at": "2018-04-09T15:12:28.720Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16430674) by Phillip Cloud (cpcloud):*\n`[~xhochy]` Is this fixed?"
        },
        {
            "created_at": "2018-04-09T15:13:27.459Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16430679) by Uwe Korn (uwe):*\nYes it is."
        },
        {
            "created_at": "2018-06-24T06:01:33.946Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16521382) by Wes McKinney (wesm):*\n`[~pi]` can you try out one of the dev wheels from https://github.com/kszucs/crossbow/releases and see if it works for you on your platform? Then we can resolve this issue"
        },
        {
            "created_at": "2018-06-24T08:49:21.739Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16521450) by Paul Ivanov (pi):*\nSorry if I'm being dense and I should do something else to install a wheel directly, but I just tried the following with Python 3.6 and did not succeed:\r\n\r\n\u00a0\r\n\r\n\u00a0\r\n```java\n\r\n$ pip install pyarrow-0.9.1.dev315-py36_0-osx-64.tar.bz2\r\nProcessing ./pyarrow-0.9.1.dev315-py36_0-osx-64.tar.bz2\r\n\u00a0 \u00a0 Complete output from command python setup.py egg_info:\r\n\u00a0 \u00a0 Traceback (most recent call last):\r\n\u00a0 \u00a0 \u00a0 File \"<string>\", line 1, in <module>\r\n\u00a0 \u00a0 \u00a0 File \"/Users/pi/.virtualenvs/try-arrow-0624/lib/python3.6/tokenize.py\", line 452, in open\r\n\u00a0 \u00a0 \u00a0 \u00a0 buffer = _builtin_open(filename, 'rb')\r\n\u00a0 \u00a0 FileNotFoundError: [Errno 2] No such file or directory: '/private/var/folders/jd/xy96sqhd0pqcb9yc7ykn5q500000gn/T/pip-req-build-eqbrjpsq/setup.py'\u00a0\u00a0\r\n\u00a0 \u00a0----------------------------------------\r\nCommand \"python setup.py egg_info\" failed with error code 1 in /private/var/folders/jd/xy96sqhd0pqcb9yc7ykn5q500000gn/T/pip-req-build-eqbrjpsq/\r\n```\r\n\u00a0\r\n\r\nThe error on Python 2.7 looks similar:\r\n\r\n\u00a0\r\n```java\n\r\n$ pip install ./pyarrow-0.9.1.dev315-py27_0-osx-64.tar.bz2\r\nProcessing ./pyarrow-0.9.1.dev315-py27_0-osx-64.tar.bz2\r\n\u00a0 \u00a0 Complete output from command python setup.py egg_info:\r\n\u00a0 \u00a0 Traceback (most recent call last):\r\n\u00a0 \u00a0 \u00a0 File \"<string>\", line 1, in <module>\r\n\u00a0 \u00a0 IOError: [Errno 2] No such file or directory: '/private/var/folders/jd/xy96sqhd0pqcb9yc7ykn5q500000gn/T/pip-req-build-NZO2kB/setup.py'\r\n\u00a0\u00a0 \u00a0\u00a0 \u00a0 ----------------------------------------\r\nCommand \"python setup.py egg_info\" failed with error code 1 in /private/var/folders/jd/xy96sqhd0pqcb9yc7ykn5q500000gn/T/pip-req-build-NZO2kB/\r\n```\r\n\u00a0"
        },
        {
            "created_at": "2018-06-24T18:51:22.236Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16521575) by Uwe Korn (uwe):*\n`[~pi]` You're trying to install the conda packages with pip. This won't work. You need to install e.g. https://github.com/kszucs/crossbow/releases/download/build-131/pyarrow-0.9.1.dev315-cp36-cp36m-macosx_10_6_intel.whl for Python 3.6"
        },
        {
            "created_at": "2018-07-06T16:39:12.323Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16535094) by Wes McKinney (wesm):*\n`[~pi]` could you give this another try and let us know?"
        },
        {
            "created_at": "2018-07-10T15:32:41.173Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16538784) by Paul Ivanov (pi):*\nFirst - a ton of apologies for my user error, `[~xhochy]` and `[~wesmckinn]`:\u00a0that's what I get for trying things at 2 in the morning. And sorry it took so long, I didn't have access to this machine for a while.\u00a0Unfortunately, I'm still seeing the same behavior as before:\u00a0\r\n```java\n\r\npip install https://github.com/kszucs/crossbow/releases/download/build-163/pyarrow-0.9.1.dev325-cp36-cp36m-macosx_10_6_intel.whl\r\nCollecting pyarrow==0.9.1.dev325 from https://github.com/kszucs/crossbow/releases/download/build-163/pyarrow-0.9.1.dev325-cp36-cp36m-macosx_10_6_intel.whl\r\nDownloading https://github.com/kszucs/crossbow/releases/download/build-163/pyarrow-0.9.1.dev325-cp36-cp36m-macosx_10_6_intel.whl (8.1MB)\r\n100% |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 8.1MB 481kB/s \r\nCollecting six>=1.0.0 (from pyarrow==0.9.1.dev325)\r\nUsing cached https://files.pythonhosted.org/packages/67/4b/141a581104b1f6397bfa78ac9d43d8ad29a7ca43ea90a2d863fe3056e86a/six-1.11.0-py2.py3-none-any.whl\r\nCollecting numpy>=1.10 (from pyarrow==0.9.1.dev325)\r\nUsing cached https://files.pythonhosted.org/packages/f6/cd/b2c50b5190b66c711c23ef23c41d450297eb5a54d2033f8dcb3b8b13ac85/numpy-1.14.5-cp36-cp36m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl\r\nInstalling collected packages: six, numpy, pyarrow\r\nSuccessfully installed numpy-1.14.5 pyarrow-0.9.1.dev325 six-1.11.0\r\n```\r\n\u00a0and the result is still\u00a0\r\n```java\n\r\n$ python\r\nPython 3.6.1 (v3.6.1:69c0db5050, Mar 21 2017, 01:21:04) \r\n[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin\r\nType \"help\", \"copyright\", \"credits\" or \"license\" for more information.\r\n>>> import pyarrow\r\nTraceback (most recent call last):\r\nFile \"<stdin>\", line 1, in <module>\r\nFile \"/Users/pi/.virtualenvs/arrow-2018-07-10/lib/python3.6/site-packages/pyarrow/__init__.py\", line 47, in <module>\r\nfrom pyarrow.lib import cpu_count, set_cpu_count\r\nImportError: dlopen(/Users/pi/.virtualenvs/arrow-2018-07-10/lib/python3.6/site-packages/pyarrow/lib.cpython-36m-darwin.so, 2): Library not loaded: @rpath/libarrow.10.dylib\r\nReferenced from: /Users/pi/.virtualenvs/arrow-2018-07-10/lib/python3.6/site-packages/pyarrow/lib.cpython-36m-darwin.so\r\nReason: image not found\r\n```\r\n\u00a0\r\n\r\n\u00a0\r\n\r\n\u00a0"
        },
        {
            "created_at": "2018-07-10T17:10:06.732Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16538963) by Wes McKinney (wesm):*\nI wonder if this could be caused by https://issues.apache.org/jira/browse/ARROW-2553. Is there any way you could provide us access to the 10.9 machine? Or is there some way to run a 10.9 VM?"
        },
        {
            "created_at": "2018-07-10T17:21:54.114Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16538982) by Paul Ivanov (pi):*\nI have this machine with my at SciPy this week. Didn't bring the charger with me today, but will do that tomorrow. Otherwise I could probably set up temporary SSH for a few of you."
        },
        {
            "created_at": "2018-07-18T10:46:27.413Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16547682) by Krisztian Szucs (kszucs):*\n`[~pi]` Could You try to install https://github.com/kszucs/crossbow/releases/download/build-231-wheel-osx-cp36m/pyarrow-0.9.1.dev375-cp36-cp36m-macosx_10_6_intel.whl ?"
        },
        {
            "created_at": "2018-07-18T21:49:02.368Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16548444) by Wes McKinney (wesm):*\nI was able to boot up OS 10.9 in VMWare and reproduce the error:\r\n\r\n```Java\n\r\nAdmins-Mac:~ admin$ pip3.6 install https://github.com/kszucs/crossbow/releases/download/build-231-wheel-osx-cp36m/pyarrow-0.9.1.dev375-cp36-cp36m-macosx_10_6_intel.whl \r\nCollecting pyarrow==0.9.1.dev375 from https://github.com/kszucs/crossbow/releases/download/build-231-wheel-osx-cp36m/pyarrow-0.9.1.dev375-cp36-cp36m-macosx_10_6_intel.whl\r\n  Downloading https://github.com/kszucs/crossbow/releases/download/build-231-wheel-osx-cp36m/pyarrow-0.9.1.dev375-cp36-cp36m-macosx_10_6_intel.whl (8.4MB)\r\n    100% |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 8.4MB 994kB/s \r\nCollecting six>=1.0.0 (from pyarrow==0.9.1.dev375)\r\n  Downloading https://files.pythonhosted.org/packages/67/4b/141a581104b1f6397bfa78ac9d43d8ad29a7ca43ea90a2d863fe3056e86a/six-1.11.0-py2.py3-none-any.whl\r\nCollecting numpy>=1.10 (from pyarrow==0.9.1.dev375)\r\n  Downloading https://files.pythonhosted.org/packages/f6/cd/b2c50b5190b66c711c23ef23c41d450297eb5a54d2033f8dcb3b8b13ac85/numpy-1.14.5-cp36-cp36m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl (4.7MB)\r\n    100% |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 4.7MB 2.1MB/s \r\nInstalling collected packages: six, numpy, pyarrow\r\nSuccessfully installed numpy-1.14.5 pyarrow-0.9.1.dev375 six-1.11.0\r\nAdmins-Mac:~ admin$ python3.6\r\nPython 3.6.6 (v3.6.6:4cf1f54eb7, Jun 26 2018, 19:50:54) \r\n[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)] on darwin\r\nType \"help\", \"copyright\", \"credits\" or \"license\" for more information.\r\n>>> import pyarrow\r\nTraceback (most recent call last):\r\n  File \"<stdin>\", line 1, in <module>\r\n  File \"/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/__init__.py\", line 58, in <module>\r\n    from pyarrow.lib import cpu_count, set_cpu_count\r\nImportError: dlopen(/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/lib.cpython-36m-darwin.so, 2): Library not loaded: @rpath/libarrow.10.dylib\r\n  Referenced from: /Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/lib.cpython-36m-darwin.so\r\n  Reason: image not found\r\n```"
        },
        {
            "created_at": "2018-07-18T21:54:17.321Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16548453) by Krisztian Szucs (kszucs):*\nBad news, moreover we cannot test it on travis https://docs.travis-ci.com/user/reference/osx#OS-X-Version"
        },
        {
            "created_at": "2018-07-18T22:11:13.929Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16548466) by Wes McKinney (wesm):*\nHere's what otool shows for libarrow.dylib:\r\n\r\n```Java\n\r\n$ otool -L /Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libarrow.10.dylib \r\n/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libarrow.10.dylib:\r\n\t@rpath/libarrow.10.dylib (compatibility version 10.0.0, current version 10.0.0)\r\n\t@rpath/libarrow_boost_system.dylib (compatibility version 0.0.0, current version 0.0.0)\r\n\t@rpath/libarrow_boost_filesystem.dylib (compatibility version 0.0.0, current version 0.0.0)\r\n\t/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 307.5.0)\r\n\t/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.50.2)\r\n```\r\n\r\nIf I load the shared libraries manually things do work\r\n\r\n```Java\n\r\n>>> import ctypes                                                                                    \r\n>>> ctypes.CDLL('/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libarrow_boost_system.dylib')\r\n<CDLL '/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libarrow_boost_system.dylib', handle 7fa101d01ea0 at 0x10a795be0>\r\n>>> ctypes.CDLL('/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libarrow_boost_filesystem.dylib')\r\n<CDLL '/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libarrow_boost_filesystem.dylib', handle 7fa101c56a10 at 0x10a795c18>\r\n>>> ctypes.CDLL('/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libarrow_boost_regex.dylib')\r\n<CDLL '/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libarrow_boost_regex.dylib', handle 7fa101c66a50 at 0x10a795be0>\r\n>>> ctypes.CDLL('/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libarrow.10.dylib')\r\n<CDLL '/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libarrow.10.dylib', handle 7fa101d093e0 at 0x10a795c18>\r\n>>> ctypes.CDLL('/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libarrow_python.dylib')\r\n<CDLL '/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libarrow_python.dylib', handle 7fa101c66970 at 0x10a795be0>\r\n>>> ctypes.CDLL('/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libparquet.dylib')\r\n<CDLL '/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libparquet.dylib', handle 7fa101c66e60 at 0x10a795c18>\r\n>>> ctypes.CDLL('/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libplasma.dylib')\r\n<CDLL '/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libplasma.dylib', handle 7fa101d13260 at 0x10a795be0>\r\n>>> import pyarrow\r\nWORKS\r\n```\r\n\r\nI suspect we need to use `install_name_tool` to \"fix\" these shared libraries. \r\n\r\nOne possible helpful reference http://thecourtsofchaos.com/2013/09/16/how-to-copy-and-relink-binaries-on-osx/\r\n"
        },
        {
            "created_at": "2018-07-18T22:13:38.545Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16548468) by Wes McKinney (wesm):*\nHere are all the loader paths fwiw\r\n\r\n```Java\n\r\n$ otool -L /Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/lib.cpython-36m-darwin.so \r\n/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/lib.cpython-36m-darwin.so:\r\n\t@rpath/libarrow.10.dylib (compatibility version 10.0.0, current version 10.0.0)\r\n\t@rpath/libarrow_python.10.dylib (compatibility version 10.0.0, current version 10.0.0)\r\n\t@rpath/libparquet.1.dylib (compatibility version 1.0.0, current version 1.4.1)\r\n\t@rpath/libplasma.10.dylib (compatibility version 10.0.0, current version 10.0.0)\r\n\t/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 307.5.0)\r\n\t/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.50.2)\r\nAdmins-Mac:~ admin$ otool -L /Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libarrow.10.dylib \r\n/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libarrow.10.dylib:\r\n\t@rpath/libarrow.10.dylib (compatibility version 10.0.0, current version 10.0.0)\r\n\t@rpath/libarrow_boost_system.dylib (compatibility version 0.0.0, current version 0.0.0)\r\n\t@rpath/libarrow_boost_filesystem.dylib (compatibility version 0.0.0, current version 0.0.0)\r\n\t/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 307.5.0)\r\n\t/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.50.2)\r\nAdmins-Mac:~ admin$ otool -L /Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libplasma.10.dylib \r\n/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libplasma.10.dylib:\r\n\t@rpath/libplasma.10.dylib (compatibility version 10.0.0, current version 10.0.0)\r\n\t@rpath/libarrow_boost_system.dylib (compatibility version 0.0.0, current version 0.0.0)\r\n\t@rpath/libarrow_boost_filesystem.dylib (compatibility version 0.0.0, current version 0.0.0)\r\n\t/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 307.5.0)\r\n\t/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.50.2)\r\nAdmins-Mac:~ admin$ otool -L /Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libarrow_python.10.dylib \r\n/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libarrow_python.10.dylib:\r\n\t@rpath/libarrow_python.10.dylib (compatibility version 10.0.0, current version 10.0.0)\r\n\t@rpath/libarrow.10.dylib (compatibility version 10.0.0, current version 10.0.0)\r\n\t/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.50.2)\r\n\t/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1349.64.0)\r\n\t/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 307.5.0)\r\nAdmins-Mac:~ admin$ otool -L /Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/lib.cpython-36m-darwin.so \r\n/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/lib.cpython-36m-darwin.so:\r\n\t@rpath/libarrow.10.dylib (compatibility version 10.0.0, current version 10.0.0)\r\n\t@rpath/libarrow_python.10.dylib (compatibility version 10.0.0, current version 10.0.0)\r\n\t@rpath/libparquet.1.dylib (compatibility version 1.0.0, current version 1.4.1)\r\n\t@rpath/libplasma.10.dylib (compatibility version 10.0.0, current version 10.0.0)\r\n\t/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 307.5.0)\r\n\t/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.50.2)\r\n```\r\n\r\nand for boost_filesystem\r\n\r\n```Java\n\r\n$ otool -L /Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libarrow_boost_filesystem.dylib \r\n/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pyarrow/libarrow_boost_filesystem.dylib:\r\n\t@rpath/libarrow_boost_filesystem.dylib (compatibility version 0.0.0, current version 0.0.0)\r\n\t@rpath/libarrow_boost_system.dylib (compatibility version 0.0.0, current version 0.0.0)\r\n\t/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 307.5.0)\r\n\t/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1238.60.2)\r\n```\r\n\r\nThis rpath stuff is really confusing, and I can't find much helpful stuff online about it"
        },
        {
            "created_at": "2018-07-18T22:26:35.820Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16548495) by Wes McKinney (wesm):*\nHere's a thread from macports packaging LLVM\r\n\r\nhttps://trac.macports.org/ticket/51542\r\n\r\nIn the changeset the fix was to add ` -Wl,-rpath,@loader_path/` to ldflags\r\n\r\nI ran the following commands:\r\n\r\n```Java\n\r\nAdmins-Mac:pyarrow admin$ install_name_tool -add_rpath @loader_path/ libarrow_boost_system.\r\nlibarrow_boost_system.1.65.1.dylib  libarrow_boost_system.dylib\r\nAdmins-Mac:pyarrow admin$ install_name_tool -add_rpath @loader_path/ libarrow_boost_system.dylib \r\nAdmins-Mac:pyarrow admin$ install_name_tool -add_rpath @loader_path/ libarrow_boost_filesystem.dylib  \r\nAdmins-Mac:pyarrow admin$ install_name_tool -add_rpath @loader_path/ libarrow_boost_regex.dylib \r\nAdmins-Mac:pyarrow admin$ install_name_tool -add_rpath @loader_path/ libarrow_python.dylib \r\nAdmins-Mac:pyarrow admin$ install_name_tool -add_rpath @loader_path/ libarrow.dylib \r\nAdmins-Mac:pyarrow admin$ install_name_tool -add_rpath @loader_path/ libparquet.dylib \r\nAdmins-Mac:pyarrow admin$ install_name_tool -add_rpath @loader_path/ libplasma.dylib \r\nAdmins-Mac:pyarrow admin$ install_name_tool -add_rpath @loader_path/ lib.cpython-36m-darwin.so \r\n```\r\n\r\nand then `import pyarrow` works."
        },
        {
            "created_at": "2018-07-18T22:27:55.141Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16548499) by Wes McKinney (wesm):*\nUpdate, tried with a fresh wheel and this command was sufficient to solve the problem:\r\n\r\n```Java\n\r\ninstall_name_tool -add_rpath @loader_path/ lib.cpython-36m-darwin.so \r\n```\r\n\r\nSo it seems we only need to fix the Cython extensions"
        },
        {
            "created_at": "2018-07-18T22:30:24.381Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16548504) by Wes McKinney (wesm):*\nIn theory this should be set by https://github.com/apache/arrow/blob/master/python/CMakeLists.txt#L460 but I'm not sure what's going wrong."
        },
        {
            "created_at": "2018-07-18T22:41:37.731Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16548512) by Wes McKinney (wesm):*\n`[~kszucs]` if you could get the macOS build to print the compile and link commands when building the Cython modules it would help to debug; they aren't being shown:\r\n\r\nhttps://travis-ci.org/kszucs/crossbow/builds/405487529#L6127"
        },
        {
            "created_at": "2018-07-18T22:53:26.031Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16548536) by Wes McKinney (wesm):*\nThis is all that's needed I think https://github.com/apache/arrow/pull/2289"
        },
        {
            "created_at": "2018-07-18T23:07:09.624Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16548554) by Krisztian Szucs (kszucs):*\nIt looks like We've created 2 PRs: https://github.com/apache/arrow/pull/2290\r\nPending Build: https://travis-ci.org/kszucs/crossbow/builds/405580361 "
        },
        {
            "created_at": "2018-07-19T00:38:06.532Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16548622) by Wes McKinney (wesm):*\nI made this a blocker. Now that I'm able to test myself we should endeavor to fix this prior to release"
        },
        {
            "created_at": "2018-07-19T01:10:41.784Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16548639) by Wes McKinney (wesm):*\nFor historical purposes, here's the offending linker call \r\n\r\n```Java\n\r\n/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++  -std=c++11  -Wall -Wno-unknown-warning-option -stdlib=libc++ -fPIC -arch x86_64  -arch x86_64 -std=c++11 -Qunused-arguments -O3 -DNDEBUG  -fno-omit-frame-pointer -Wno-unused-variable -Wno-maybe-uninitialized -Wno-parentheses-equality -Wno-constant-logical-operand -Wno-missing-declarations -Wno-return-type-c-linkage -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -mmacosx-version-min=10.9 -bundle -Wl,-headerpad_max_install_names -undefined dynamic_lookup -o release/lib.cpython-36m-darwin.so CMakeFiles/lib.dir/lib.cpp.o -Wl,-rpath,@loader_path /Users/travis/build/kszucs/crossbow/arrow/arrow-dist/lib/libarrow.dylib /Users/travis/build/kszucs/crossbow/arrow/arrow-dist/lib/libarrow_python.dylib /Users/travis/build/kszucs/crossbow/arrow/arrow-dist/lib/libparquet.dylib /Users/travis/build/kszucs/crossbow/arrow/arrow-dist/lib/libplasma.dylib \r\n```\r\n\r\nNote the `-Wl,-rpath,@loader_path`. If somehow `-Wl,-rpath,@loader_path/` works but `-Wl,-rpath,@loader_path` doesn't I'll be pretty flabberghasted"
        },
        {
            "created_at": "2018-07-19T07:49:30.779Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-2326?focusedCommentId=16548935) by Uwe Korn (uwe):*\nIssue resolved by pull request 2291\n<https://github.com/apache/arrow/pull/2291>"
        }
    ]
}